modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -28,16 +28,26 @@
              [i18n :refer [tru]]])
   (:import [java.sql ResultSet Types]
            [java.time OffsetDateTime ZonedDateTime]
-           metabase.util.honeysql_extensions.Identifier
-           net.snowflake.client.jdbc.SnowflakeSQLException))
+           metabase.util.honeysql_extensions.Identifier))
 
 (driver/register! :snowflake, :parent #{:sql-jdbc ::legacy/use-legacy-classes-for-read-and-set})
 
+(defmethod driver/humanize-connection-error-message :snowflake
+  [_ message]
+  (log/spy :error (type message))
+  (condp re-matches message
+    #"(?s).*Object does not exist.*$"
+    (driver.common/connection-error-messages :database-name-incorrect)
+
+    #"(?s).*" ; default - the Snowflake errors have a \n in them
+    message))
+
 (defmethod sql-jdbc.conn/connection-details->spec :snowflake
   [_ {:keys [account regionid], :as opts}]
   (let [host (if regionid
                (str account "." regionid)
-               account)]
+               account)
+        upcase-not-nil (fn [s] (when s (str/upper-case s)))]
```

Comment:
I think this is going to be a problem with Turkish

Reply:
hah yes you're right, let me go change that